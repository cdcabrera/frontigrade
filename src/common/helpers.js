import _ from 'lodash';
import moment from 'moment/moment';

const copyClipboard = text => {
  let successful;

  try {
    window.getSelection().removeAllRanges();

    const newTextarea = document.createElement('pre');
    newTextarea.appendChild(document.createTextNode(text));

    newTextarea.style.position = 'absolute';
    newTextarea.style.top = '-1000px';
    newTextarea.style.left = '-1000px';
    newTextarea.style.overflow = 'hidden';
    newTextarea.style.width = '1px';
    newTextarea.style.height = '1px';

    const range = document.createRange();
    window.document.body.appendChild(newTextarea);

    range.selectNode(newTextarea);

    window.getSelection().addRange(range);

    successful = window.document.execCommand('copy');

    window.document.body.removeChild(newTextarea);
    window.getSelection().removeAllRanges();
  } catch (e) {
    successful = false;
    console.warn('Copy to clipboard failed.', e.message);
  }

  return successful;
};

const generateId = prefix => `${prefix || 'generatedid'}-${Math.ceil(1e5 * Math.random())}`;

const generateHoursFromSeconds = seconds => {
  let parsedSeconds = Number.parseFloat(seconds);
  parsedSeconds = Number.isNaN(parsedSeconds) ? null : parsedSeconds;

  parsedSeconds =
    parsedSeconds === null ? parsedSeconds : Math.floor(moment.duration(parsedSeconds, 'seconds').asHours());

  return parsedSeconds;
};

const generatePriorYearMonthArray = () => {
  const dateStart = moment().subtract(12, 'months');
  const dateEnd = moment().startOf('month');

  let timeValues = [];

  while (dateEnd > dateStart || dateStart.format('M') === dateEnd.format('M')) {
    timeValues.push({
      title: dateStart.startOf('month').format('YYYY MMMM'),
      start: dateStart.startOf('month').toISOString(),
      end: dateStart.endOf('month').toISOString()
    });

    dateStart.add(1, 'month');
  }

  timeValues = timeValues.reverse();
  timeValues[0] = {
    default: true,
    title: 'Last 30 Days',
    start: moment()
      .startOf('day')
      .subtract(30, 'days')
      .toISOString(),
    end: moment()
      .endOf('day')
      .toISOString()
  };

  return { defaultTime: timeValues[0], timeValues };
};

const getMessageFromResults = (results, filterField = null) => {
  const status = _.get(results, 'response.status', results.status);
  const statusResponse = _.get(results, 'response.statusText', results.statusText);
  const messageResponse = _.get(results, 'response.data', results.message);
  const detailResponse = _.get(results, 'response.data', results.detail);

  if (status < 400 && !messageResponse && !detailResponse) {
    return statusResponse;
  }

  if (status >= 500 || (status === undefined && (messageResponse || detailResponse))) {
    return `${status || ''} Server is currently unable to handle this request. ${messageResponse ||
      detailResponse ||
      ''}`;
  }

  if (typeof messageResponse === 'string') {
    return messageResponse;
  }

  if (typeof detailResponse === 'string') {
    return detailResponse;
  }

  const getMessages = (messageObject, filterKey) => {
    const obj = filterKey ? messageObject[filterKey] : messageObject;

    return _.map(
      obj,
      next => {
        if (_.isArray(next)) {
          return getMessages(next);
        }

        return next;
      },
      null
    );
  };

  return _.join(getMessages(messageResponse || detailResponse, filterField), '\n');
};

const getStatusFromResults = results => {
  let status = _.get(results, 'response.status', results.status);

  if (status === undefined) {
    status = 0;
  }

  return status;
};

const noop = Function.prototype;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};

const prettyPrintJson = json => JSON.stringify(json, null, 2);

const DEV_MODE = process.env.REACT_APP_ENV === 'development';

const OC_MODE = process.env.REACT_APP_ENV === 'oc';

const RH_BRAND = process.env.REACT_APP_RH_BRAND === 'true';

const UI_COMMIT_HASH = process.env.REACT_APP_UI_COMMIT_HASH;

const FULFILLED_ACTION = (base = '') => `${base}_FULFILLED`;

const PENDING_ACTION = (base = '') => `${base}_PENDING`;

const REJECTED_ACTION = (base = '') => `${base}_REJECTED`;

const HTTP_STATUS_RANGE = status => `${status}_STATUS_RANGE`;

const helpers = {
  copyClipboard,
  generateId,
  generateHoursFromSeconds,
  generatePriorYearMonthArray,
  getMessageFromResults,
  getStatusFromResults,
  noop,
  setStateProp,
  prettyPrintJson,
  DEV_MODE,
  OC_MODE,
  RH_BRAND,
  UI_COMMIT_HASH,
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION,
  HTTP_STATUS_RANGE
};

export { helpers as default, helpers };
