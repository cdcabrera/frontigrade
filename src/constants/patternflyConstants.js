import { BackgroundImageSrc } from '@patternfly/react-core';
import bgFilter from '@patternfly/patternfly-next/assets/images/background-filter.svg';
import pfBg576 from '@patternfly/patternfly-next/assets/images/pfbg_576.jpg';
import pfBg576R2x from '@patternfly/patternfly-next/assets/images/pfbg_576@2x.jpg';
import pfBg768 from '@patternfly/patternfly-next/assets/images/pfbg_768.jpg';
import pfBg768R2x from '@patternfly/patternfly-next/assets/images/pfbg_768@2x.jpg';
import pfBg992 from '@patternfly/patternfly-next/assets/images/pfbg_992.jpg';
import pfBg992R2x from '@patternfly/patternfly-next/assets/images/pfbg_992@2x.jpg';
import pfBg1200 from '@patternfly/patternfly-next/assets/images/pfbg_1200.jpg';
import pfBg2000 from '@patternfly/patternfly-next/assets/images/pfbg_2000.jpg';

export const bgImages = {
  [BackgroundImageSrc.lg]: pfBg1200,
  [BackgroundImageSrc.md]: pfBg992,
  [BackgroundImageSrc.md2x]: pfBg992R2x,
  [BackgroundImageSrc.sm]: pfBg768,
  [BackgroundImageSrc.sm2x]: pfBg768R2x,
  [BackgroundImageSrc.xl]: pfBg2000,
  [BackgroundImageSrc.xs]: pfBg576,
  [BackgroundImageSrc.xs2x]: pfBg576R2x,
  [BackgroundImageSrc.filter]: `${bgFilter}#image_overlay`
};

export default bgImages;
