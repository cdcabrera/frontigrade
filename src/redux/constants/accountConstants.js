const ADD_ACCOUNT = 'ADD_ACCOUNT';
const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
const DELETE_ACCOUNTS = 'DELETES_ACCOUNTS';
const GET_ACCOUNT = 'GET_ACCOUNT';
const GET_ACCOUNTS = 'GET_ACCOUNTS';
const UPDATE_ACCOUNTS = 'UPDATE_ACCOUNTS';
const UPDATE_ACCOUNT = 'UPDATE_ACCOUNT';
const UPDATE_ACCOUNT_FIELD = 'UPDATE_ACCOUNT_FIELD';
const UPDATE_ACCOUNT_IMAGE = 'UPDATE_ACCOUNT_IMAGE';
const UPDATE_ACCOUNT_IMAGE_FIELD = 'UPDATE_ACCOUNT_IMAGE_FIELD';
const UPDATE_ACCOUNT_IMAGES_INSTANCES = 'UPDATE_ACCOUNT_IMAGES_INSTANCES';
const GET_ACCOUNT_IMAGES = 'GET_ACCOUNT_IMAGES';
const GET_ACCOUNT_IMAGES_INSTANCES = 'GET_ACCOUNT_IMAGES_INSTANCES';
const GET_ACCOUNT_INSTANCES = 'GET_ACCOUNT_INSTANCES';
// ToDo: clean up account constants, ADD_ACCOUNT_WIZARD not used
const ADD_ACCOUNT_WIZARD = 'ADD_ACCOUNT_WIZARD';
const ADD_ACCOUNT_WIZARD_STEP_POLICY = 'ADD_ACCOUNT_WIZARD_STEP_POLICY';
const INVALID_ACCOUNT_WIZARD_STEP_POLICY = 'INVALID_ACCOUNT_WIZARD_STEP_POLICY';
const ADD_ACCOUNT_WIZARD_STEP_ROLE = 'ADD_ACCOUNT_WIZARD_STEP_ROLE';
const INVALID_ACCOUNT_WIZARD_STEP_ROLE = 'INVALID_ACCOUNT_WIZARD_STEP_ROLE';
const ADD_ACCOUNT_WIZARD_STEP_ARN = 'ADD_ACCOUNT_WIZARD_STEP_ARN';
const INVALID_ACCOUNT_WIZARD_STEP_ARN = 'INVALID_ACCOUNT_WIZARD_STEP_ARN';
// ToDo: end clean up account constants, poorly named, aim at using ADD_ACCOUNT_WIZARD_ prefix
const ADD_ACCOUNT_SHOW = 'ADD_ACCOUNT_SHOW';
const UPDATE_ACCOUNT_HIDE = 'UPDATE_ACCOUNT_HIDE';

const DELETE_ACCOUNT_HIDE = 'DELETE_ACCOUNT_HIDE';
const DELETE_ACCOUNT_SHOW = 'DELETE_ACCOUNT_SHOW';
const EDIT_ACCOUNT_HIDE = 'EDIT_ACCOUNT_HIDE';
const EDIT_ACCOUNT_SHOW = 'EDIT_ACCOUNT_SHOW';

export {
  ADD_ACCOUNT,
  DELETE_ACCOUNT,
  DELETE_ACCOUNTS,
  GET_ACCOUNT,
  GET_ACCOUNTS,
  UPDATE_ACCOUNTS,
  UPDATE_ACCOUNT,
  UPDATE_ACCOUNT_FIELD,
  UPDATE_ACCOUNT_IMAGE,
  UPDATE_ACCOUNT_IMAGE_FIELD,
  UPDATE_ACCOUNT_IMAGES_INSTANCES,
  GET_ACCOUNT_IMAGES,
  GET_ACCOUNT_IMAGES_INSTANCES,
  GET_ACCOUNT_INSTANCES,
  ADD_ACCOUNT_WIZARD,
  ADD_ACCOUNT_WIZARD_STEP_POLICY,
  INVALID_ACCOUNT_WIZARD_STEP_POLICY,
  ADD_ACCOUNT_WIZARD_STEP_ROLE,
  INVALID_ACCOUNT_WIZARD_STEP_ROLE,
  ADD_ACCOUNT_WIZARD_STEP_ARN,
  INVALID_ACCOUNT_WIZARD_STEP_ARN,
  ADD_ACCOUNT_SHOW,
  UPDATE_ACCOUNT_HIDE,
  DELETE_ACCOUNT_HIDE,
  DELETE_ACCOUNT_SHOW,
  EDIT_ACCOUNT_HIDE,
  EDIT_ACCOUNT_SHOW
};
