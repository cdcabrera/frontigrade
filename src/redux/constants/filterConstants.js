const TOOLBAR_CLEAR_FILTERS = 'TOOLBAR_CLEAR_FILTERS';
const TOOLBAR_ADD_FILTER = 'TOOLBAR_ADD_FILTER';
const TOOLBAR_REMOVE_FILTER = 'TOOLBAR_REMOVE_FILTER';
const TOOLBAR_SET_DATE_TYPE = 'TOOLBAR_SET_DATE_TYPE';
const TOOLBAR_CLEAR_DATE_TYPE = 'TOOLBAR_CLEAR_DATE_TYPE';
const TOOLBAR_SET_FILTER_TYPE = 'TOOLBAR_SET_FILTER_TYPE';
const TOOLBAR_SET_FILTER_VALUE = 'TOOLBAR_SET_FILTER_VALUE';
const TOOLBAR_SET_SORT_TYPE = 'TOOLBAR_SET_SORT_TYPE';
const TOOLBAR_TOGGLE_SORT_ASCENDING = 'TOOLBAR_TOGGLE_SORT_ASCENDING';

export {
  TOOLBAR_ADD_FILTER,
  TOOLBAR_CLEAR_FILTERS,
  TOOLBAR_REMOVE_FILTER,
  TOOLBAR_SET_DATE_TYPE,
  TOOLBAR_CLEAR_DATE_TYPE,
  TOOLBAR_SET_FILTER_TYPE,
  TOOLBAR_SET_FILTER_VALUE,
  TOOLBAR_SET_SORT_TYPE,
  TOOLBAR_TOGGLE_SORT_ASCENDING
};
