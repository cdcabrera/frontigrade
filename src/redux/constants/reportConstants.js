const REPORT_EXPORT = 'REPORT_EXPORT';

export { REPORT_EXPORT as default, REPORT_EXPORT };
