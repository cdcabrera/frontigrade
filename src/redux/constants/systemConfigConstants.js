const GET_SYSTEM_CONFIG = 'GET_SYSTEM_CONFIG';

export { GET_SYSTEM_CONFIG as default, GET_SYSTEM_CONFIG };
