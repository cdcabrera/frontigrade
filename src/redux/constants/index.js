import * as accountTypes from './accountConstants';
import * as confirmationModalTypes from './confirmationModalConstants';
import * as systemConfigTypes from './systemConfigConstants';
import * as toastNotificationTypes from './toastNotificationConstants';
import * as userTypes from './userConstants';

const reduxTypes = {
  account: accountTypes,
  confirmationModal: confirmationModalTypes,
  systemConfig: systemConfigTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  accountTypes,
  confirmationModalTypes,
  systemConfigTypes,
  toastNotificationTypes,
  userTypes
};
