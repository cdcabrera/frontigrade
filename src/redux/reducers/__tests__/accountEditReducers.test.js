import { accountEditReducers } from '..';
import { accountTypes as types } from '../../constants';
import helpers from '../../../common/helpers';

describe('AccountEditReducers', () => {
  it('should return the initial state', () => {
    expect(accountEditReducers.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [
      types.DELETE_ACCOUNT_HIDE,
      types.DELETE_ACCOUNT_SHOW,
      types.EDIT_ACCOUNT_HIDE,
      types.EDIT_ACCOUNT_SHOW
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = accountEditReducers(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined error types', () => {
    Object.keys(types).forEach(value => {
      if (/wizard/i.test(value)) {
        return;
      }

      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = accountEditReducers(undefined, dispatched);

      expect({ type: helpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot('rejected types');
    });
  });

  it('should handle all defined pending types', () => {
    Object.keys(types).forEach(value => {
      if (/wizard/i.test(value)) {
        return;
      }

      const dispatched = {
        type: helpers.PENDING_ACTION(value)
      };

      const resultState = accountEditReducers(undefined, dispatched);

      expect({ type: helpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot('pending types');
    });
  });

  it('should handle all defined fulfilled types', () => {
    Object.keys(types).forEach(value => {
      if (/wizard/i.test(value)) {
        return;
      }

      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            lorem: 'success'
          }
        }
      };

      const resultState = accountEditReducers(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot('fulfilled types');
    });
  });
});
