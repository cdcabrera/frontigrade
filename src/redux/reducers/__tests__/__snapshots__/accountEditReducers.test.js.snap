// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountEditReducers should handle all defined error types: rejected types 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "DELETE_ACCOUNT_REJECTED",
}
`;

exports[`AccountEditReducers should handle all defined error types: rejected types 2`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_REJECTED",
}
`;

exports[`AccountEditReducers should handle all defined error types: rejected types 3`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": true,
      "errorMessage": "ERROR",
      "errorStatus": 0,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_FIELD_REJECTED",
}
`;

exports[`AccountEditReducers should handle all defined fulfilled types: fulfilled types 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "DELETE_ACCOUNT_FULFILLED",
}
`;

exports[`AccountEditReducers should handle all defined fulfilled types: fulfilled types 2`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_FULFILLED",
}
`;

exports[`AccountEditReducers should handle all defined fulfilled types: fulfilled types 3`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": true,
      "pending": false,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_FIELD_FULFILLED",
}
`;

exports[`AccountEditReducers should handle all defined pending types: pending types 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "DELETE_ACCOUNT_PENDING",
}
`;

exports[`AccountEditReducers should handle all defined pending types: pending types 2`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_PENDING",
}
`;

exports[`AccountEditReducers should handle all defined pending types: pending types 3`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": true,
      "show": false,
    },
  },
  "type": "UPDATE_ACCOUNT_FIELD_PENDING",
}
`;

exports[`AccountEditReducers should handle specific defined types: defined type DELETE_ACCOUNT_HIDE 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "DELETE_ACCOUNT_HIDE",
}
`;

exports[`AccountEditReducers should handle specific defined types: defined type DELETE_ACCOUNT_SHOW 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": undefined,
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": true,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "DELETE_ACCOUNT_SHOW",
}
`;

exports[`AccountEditReducers should handle specific defined types: defined type EDIT_ACCOUNT_HIDE 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
  },
  "type": "EDIT_ACCOUNT_HIDE",
}
`;

exports[`AccountEditReducers should handle specific defined types: defined type EDIT_ACCOUNT_SHOW 1`] = `
Object {
  "result": Object {
    "del": Object {
      "account": Object {},
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": false,
    },
    "modal": Object {
      "account": undefined,
      "error": false,
      "errorMessage": null,
      "errorStatus": null,
      "fulfilled": false,
      "pending": false,
      "show": true,
    },
  },
  "type": "EDIT_ACCOUNT_SHOW",
}
`;
